lib/retry.ts

```diff
@@ -75,26 +75,23 @@ export interface RetryConfig<T> {
  * @ignore
  */
 function validateRetryConfig<T>(config: RetryConfig<T>): void {
-  if (!config.operation || typeof config.operation !== "function") {
-    throw new Error("'operation' is a required property and must be of type 'function' " +
-      "that returns a Promise.");
-  }
+  throwTypeErrorIfParameterMissing("operation", config.operation);
+  throwTypeErrorIfParameterTypeMismatch("operation", config.operation, "function");
 
-  if (!config.connectionId || typeof config.connectionId !== "string") {
-    throw new Error("'connectionId' is a required property and must be of type 'string'.");
-  }
+  throwTypeErrorIfParameterMissing("connectionId", config.connectionId);
 
-  if (!config.operationType || typeof config.operationType !== "string") {
-    throw new Error("'operationType' is a required property and must be of type 'string'.");
-  }
+  throwTypeErrorIfParameterMissing("operationType", config.operationType);
+  throwTypeErrorIfParameterTypeMismatch("operationType", config.operationType, "string");
 
-  if (config.times && typeof config.times !== "number") {
-    throw new Error("'times' must be of type 'number'.");
+
+  if (config.times) {
+    throwTypeErrorIfParameterTypeMismatch("times", config.times, "number");
   }
 
-  if (config.delayInSeconds && typeof config.delayInSeconds !== "number") {
-    throw new Error("'delayInSeconds' must be of type 'number'.");
+  if (config.delayInSeconds) {
+    throwTypeErrorIfParameterTypeMismatch("delayInSeconds", config.delayInSeconds, "number");
```

Comment:
We should be able to remove all of the above checks.

Type checks are useful if are checking end user input.
`retry` is something that will not get inputs dynamically.

Reply:
>We should be able to remove all of the above checks.

All the type checks?

>Type checks are useful if are checking end user input.
retry is something that will not get inputs dynamically.

Do you think if user will use this library for different service then we should give the type error?

Reply:
This is for the `retry` method which will not get dynamic input with varying types. We usually add type checks if the function is user facing and can get dynamic inputs.